### PR TITLE
add documentation for query and filters, tweak some css for fonts, heading etc. 

### PR DESCRIPTION
**Pull request by himank**

### CODE REVIEW

**Review by adilansari — APPROVED** [review]

File: src/css/custom.css [hunk 1]
```diff
@@ -15,6 +19,32 @@
   --ifm-color-primary-lightest: #5ecbad;
 }
 
+h1 {
```
Comment (ovaistariq) [inline src/css/custom.css:22]: Let's not make the font changes like this. It we need to change the fonts we can revisit it later. The font size changes, for example, have to be done so font sizing is dynamic depending on the size of the screen.

File: docs/overview/query.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
+# Querying Documents
+
+Tigris provides powerful query functionality for specifying which documents you want to retrieve. There is no need to
+index any field as the Tigris allows querying on any field of a document.
```
Comment (ovaistariq) [inline docs/overview/query.md:4]: index any field as Tigris ..

File: docs/overview/query.md [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+Even after applying multiple AND conditions, what if you need something even more complex? What about reading documents
+where we need a logical `OR` on `brand` but also need to apply logical `AND` on some other fields. Let's read products where the
+`brand` is either "adidas" or "coach" but the `price` should be less than 50 and the product should be `popular`.
+
+```shell
+curl 'http://localhost:8081/api/v1/databases/catalogdb/collections/catalog/documents/read'
+-X POST
+-H "Content-Type: application/json"
+-d '{
+   "filter": {"$OR": [{"brand": "adidas"}, {"brand": "coach"}], price": {"$lt": 50}, "popularity": {"$gte": 8}}
```
Comment (ovaistariq) [inline docs/overview/query.md:120]: better to break it into multi-line json

File: docs/typescript/filter.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
+# Filters
+
+Tigris offers powerful apis to read or search the data. These APIs support filters to add specific criteria that the
```
Comment (ovaistariq) [inline docs/typescript/filter.md:3]: Uppercase APIs

File: docs/golang/filter.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
+# Filters
```
Comment (ovaistariq) [inline docs/golang/filter.md:1]: Since filters are not going to used out of "querying", merge this into the "querying" doc.

Comment (ovaistariq) [reply docs/golang/filter.md:1]: Do the same for Java and TS as well

File: docs/golang/filter.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+Tigris offers powerful apis to read or search the data. These APIs support filters to add specific criteria that the
```
Comment (ovaistariq) [inline docs/golang/filter.md:3]: Then once you merge it, we don't need lines 3-7

File: docs/java/query.md [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+| --- | -------------- | ----- | ------------ | -------- | ---------- | ----------------------------------- |
+| 1   | fiona handbag  | 99.9  | michael kors | purses   | 8          | {"author": "alice", "rating": 7}    |
+| 2   | tote bag       | 49    | coach        | handbags | 9          | {"author": "olivia", "rating": 8.3} |
+| 3   | sling bag      | 75    | coach        | purses   | 9          | {"author": "alice", "rating": 9.2}  |
+| 4   | sneakers shoes | 40    | adidas       | shoes    | 10         | {"author": "olivia", "rating": 9}   |
+| 5   | running shoes  | 89    | nike         | shoes    | 10         | {"author": "olivia", "rating": 8.5} |
+
+The first step is to create the collection object.
+
+```shell
```
Comment (ovaistariq) [inline docs/java/query.md:20]:
```java

File: src/css/custom.css (hunk 1 above)
Comment (himank) [reply src/css/custom.css:22]: Okay, I can remove it for now but let's revisit it. I feel the font sizes are big.

File: docs/golang/filter.md (hunk 5 above)
Comment (himank) [reply docs/golang/filter.md:1]: Not only for querying, but filters are also going to be used for search as well. For Update/Delete as well syntax is the same but with limited capabilities.

Comment (ovaistariq) [reply docs/golang/filter.md:1]: Search and Querying are different functionalities and filtering for them mean different things. It is better to keep them separate

File: docs/golang/query.md [hunk 8]
```diff
@@ -0,0 +1,150 @@
+# Querying Documents
```
Comment (ovaistariq) [inline docs/golang/query.md:1]: There is a CRUD section which also covers part of the read operations. And then rest of read operations are covered here. How should we handle that?

Comment (himank) [reply docs/golang/query.md:1]: probably no need to Change the naming, we can probably say in the read something along these lines,
```
To read documents, use the `findOne`, `findMany()` or `findManyStream()` APIs. The APIs can be used as follows,

..example..

To learn more about the read capabilities, see the following guide on Querying.
```
basically point them to the detailed Querying guide. The other option is to break CRUD into individual operations like insert.md, replace.md but probably not much content.

File: docs/golang/filter.md (hunk 5 above)
Comment (ovaistariq) [reply docs/golang/filter.md:1]: What I mean is that filtering means different things in context of search and in context of CRUD operations. In context of search is means "search for all the documents by the term, and out of the documents you have found and only then apply the filtering condition". In the context of a database query the meaning if simply "find me everything that matches the filter".

Comment (himank) [reply docs/golang/filter.md:1]: That part I understand but I was mainly talking about pointing Querying/Search both to the filter syntax. But let me move it to Query for now and then the same high-level syntax probably we can add in the Search documentation as well.

**Pull request merged**